internal/token/jwt_test.go

```diff
@@ -0,0 +1,124 @@
+package token
+
+import (
+	"errors"
+	"github.com/benbjohnson/clock"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/onsi/gomega"
+	"kg/procurement/cmd/config"
+	"testing"
+	"time"
+)
+
+func Test_newJWTManager(t *testing.T) {
+	_ = newJWTManager(config.Token{}, nil)
+}
+
+func Test_GenerateToken(t *testing.T) {
+	t.Parallel()
+
+	var (
+		mockClock *clock.Mock
+		jwtMgr    *jwtManager
+	)
+
+	setup := func(t *testing.T) *gomega.GomegaWithT {
+		mockClock = clock.NewMock()
+
+		jwtMgr = &jwtManager{
+			cfg:   config.Token{Secret: "secret"},
+			clock: mockClock,
+		}
+
+		return gomega.NewWithT(t)
+	}
+
+	t.Run("GenerateTokenWithValidClaimsReturnsTokenSuccessfully", func(t *testing.T) {
```

Comment:
nit: since the test is wrapped in `Test_GenerateToken`, we could rename this to `valid claim returns successfully`